core/templates/dev/head/player/AnswerClassificationService.js

```diff
@@ -97,13 +97,20 @@ oppia.factory('answerClassificationService', [
           explorationId, oldState, answer, isInEditorMode,
           interactionRulesService) {
         var deferred = $q.defer();
+        var result = null;
+        var answerGroups = oldState.interaction.answer_groups;
+        var defaultOutcome = oldState.interaction.default_outcome;
+
         if (interactionRulesService) {
-          var answerGroups = oldState.interaction.answer_groups;
-          var defaultOutcome = oldState.interaction.default_outcome;
+          result = classifyAnswer(
+            answer, answerGroups, defaultOutcome, interactionRulesService);
+          deferred.resolve(result);
+        }
 
-          deferred.resolve(classifyAnswer(
-            answer, answerGroups, defaultOutcome, interactionRulesService));
-        } else {
+        if (result === null || (result.outcome == defaultOutcome &&
+            INTERACTION_SPECS[oldState.interaction.id]
+              .is_string_classifier_trainable &&
```

Comment:
Might be nicer to declare `oldInteraction = INTERACTION_SPECS[oldState.interaction.id]` as a separate variable in order to avoid requiring a newline before the dot.